src/App.tsx

```diff
@@ -26,70 +27,130 @@ type ReorderOptions = {
   isReversed: boolean,
 };
 
-// Use this function in the render method to prepare goods
-export function getReorderedGoods(
-  goods: string[],
-  { sortType, isReversed }: ReorderOptions,
-) {
-  // To avoid the original array mutation
-  const visibleGoods = [...goods];
+export class App extends PureComponent<{}, ReorderOptions> {
+  state: ReorderOptions = {
+    isReversed: false,
+    sortType: SortType.NONE,
+  };
 
-  // Sort and reverse goods if needed
-  // eslint-disable-next-line no-console
-  console.log(sortType, isReversed);
+  sortAlphabetically = () => {
+    this.setState({
+      sortType: SortType.ALPHABET,
+    });
+  };
 
-  return visibleGoods;
-}
+  sortByLength = () => {
+    this.setState({
+      sortType: SortType.LENGTH,
+    });
+  };
 
-// DON'T save goods to the state
-// type State = {
-//   isReversed: boolean,
-//   sortType: SortType,
-// };
-
-export const App: React.FC = () => {
-  return (
-    <div className="section content">
-      <div className="buttons">
-        <button
-          type="button"
-          className="button is-info is-light"
-        >
-          Sort alphabetically
-        </button>
-
-        <button
-          type="button"
-          className="button is-success is-light"
-        >
-          Sort by length
-        </button>
-
-        <button
-          type="button"
-          className="button is-warning is-light"
-        >
-          Reverse
-        </button>
-
-        <button
-          type="button"
-          className="button is-danger is-light"
-        >
-          Reset
-        </button>
-      </div>
+  reverse = () => {
+    this.setState((prevState) => ({
+      isReversed: !prevState.isReversed,
+    }));
+  };
+
+  reset = () => {
+    this.setState({
+      isReversed: false,
+      sortType: SortType.NONE,
+    });
+  };
+
+  getReorderedGoods = (): string[] => {
+    const { sortType, isReversed } = this.state;
+    let visibleGoods = [...goodsFromServer];
+
+    switch (sortType) {
+      case SortType.ALPHABET:
+        visibleGoods.sort();
+        break;
+
+      case SortType.LENGTH:
+        visibleGoods.sort((a, b) => a.length - b.length);
+        break;
+
+      default:
+        break;
+    }
+
+    if (isReversed) {
+      visibleGoods = visibleGoods.reverse();
+    }
+
+    return visibleGoods;
+  };
+
+  render() {
+    const { sortType, isReversed } = this.state;
+    const isResetButtonVisible = sortType !== SortType.NONE || isReversed;
+    const visibleGoods = this.getReorderedGoods();
+
+    return (
+      <div className="section content">
+        <div className="buttons">
+          <button
+            type="button"
+            className={cn(
+              'button',
+              'is-info',
+              {
+                'is-light': sortType !== SortType.ALPHABET,
+              },
+            )}
```

Comment:
Remove unnecessary linebreaks

```suggestion
            className={cn('button is-info', {
                'is-light': sortType !== SortType.ALPHABET,
              },
            )}
```